android/src/main/java/com/revenuecat/purchases/hybridcommon/common.kt

```diff
@@ -194,6 +193,74 @@ fun purchasePackage(
     }
 }
 
+fun purchaseSubscriptionOption(
+    activity: Activity?,
+    productIdentifier: String,
+    optionIdentifier: String,
+    googleOldProductId: String?,
+    googleProrationMode: GoogleProrationMode?,
+    googleIsPersonalizedPrice: Boolean?,
+    onResult: OnResult
+) {
+    if (activity != null) {
+        val onReceiveStoreProducts: (List<StoreProduct>) -> Unit = { storeProducts ->
+            val allSubscriptionOptions = storeProducts.mapNotNull { storeProduct ->
+                storeProduct.subscriptionOptions?.map { Pair(storeProduct, it) }
+            }.flatten()
```

Comment:
maybe some intermediate vars to make this more readable would be good

but i'm especially bad at following these chained calls so 🤷